circular.py

```diff
@@ -10,9 +15,79 @@ def _import_from_path(module_name, file_path):
     spec.loader.exec_module(module)
 
 
-def test_circular_imports():
-    for path, _, files in os.walk("starknet_py"):
+def assert_no_circular_imports():
+    for path, _, files in os.walk(PACKAGE_NAME):
         py_files = [f for f in files if f.endswith(".py")]
-        for file_ in py_files:
-            file_path = os.path.join(path, file_)
-            _import_from_path(file_, file_path)
+        for file in py_files:
+            file_path = os.path.join(path, file)
+
+            relative_path = os.path.relpath(file_path, PACKAGE_NAME)
+            module_path_no_ext = relative_path.removesuffix(".py")
+            # Handle __init__.py files specially
+            if module_path_no_ext.endswith("__init__"):
+                module_path_no_init = module_path_no_ext.removesuffix(
+                    "__init__"
+                ).rstrip(os.sep)
+                if not module_path_no_init:  # Top-level __init__.py
+                    module_name = PACKAGE_NAME
+                else:
+                    dotted_module_path = module_path_no_init.replace(os.sep, ".")
```

Comment:
Out of curiosity, what possible case there is that the `os.sep` is different than dots?

Reply:
os separator always shows up when using `os.path.relpath`, even when running this locally with print, e.g.

<img width="573" height="210" alt="image" src="https://github.com/user-attachments/assets/217971b9-0bb5-45f6-bb71-fce9c65f59d1" />

so the replacement is indeed needed.